esp/src/eclwatch/MemberOfWidget.js

```diff
@@ -50,52 +55,135 @@ define([
+                disabled: true,
+                onClick: function (event) {
+                    context._onDeleteMember(event);
                 }
+            }).placeAt(this.addButton.domNode, "after");
+            tmpSplitter = new ToolbarSeparator().placeAt(this.addButton.domNode, "before");
+            this.dialog = new Dialog({
+                title: this.i18n.PleaseSelectAGroupToAddUser,
+                style: "width: 320px;"
             });
-            this._refreshActionState();
-        },
+            this.dialogBtn = new Button({
+                label: this.i18n.Add,
+                style: "float:right;",
+                onClick: function (event) {
+                    context._onSubmitAddMember(event);
+                }
+            }).placeAt(this.dialog.domNode);
 
-        createGrid: function (domID) {
-            var retVal = new declare([ESPUtil.Grid(true, false)])({
+            var retVal = new declare([ESPUtil.Grid(true, true)])({
                 sort: [{ attribute: "name" }],
                 store: this.store,
                 columns: {
-                    isMember: editor({
-                        label: "",
+                    check: selector({
                         width: 27,
-                        editor: "checkbox",
-                        autoSave: true
-                    }),
+                        label: " "
+                    }, "checkbox"),
                     name: {
                         label: this.i18n.GroupName,
                         sortable: true
                     }
                 }
             }, domID);
 
+            retVal.onSelectionChanged(function (event) {
+                context.refreshActionState();
+            });
+
             return retVal;
         },
 
+        _onAddMember: function (event) {
+            this.dialog.show();
+        },
+
+        _onSubmitAddMember: function (event) {
+            var context = this;
+            WsAccess.UserGroupEdit({
+                request: {
+                    groupnames_i1:context.memberDropDown.get("value"),
+                    username:context.params.username,
+                    action: "Add"
+                }
+            }).then(function (response) {
+               context.dialog.hide();
+               context._onRefresh();
+            });
+        },
+
+        _onDeleteMember: function (event) {
+            var context = this;
+            var selections = this.grid.getSelected();
+            if (confirm(this.i18n.YouAreAboutToRemoveUserFrom)) {
+                for (var i = selections.length - 1; i >= 0; --i) {
+                    WsAccess.UserGroupEdit({
+                        request: {
+                            groupnames_i6: selections[i].name,
+                            username:context.params.username,
+                            action: "Delete"
+                        }
+                    }).then(function (response) {
+                        context._onRefresh();
```

Comment:
This will cause the list to refresh once per item deleted - performance issue more than issue.

FYI there is an "all" function which lets you delete "all" the items and then have a single "then" at the end, which you can do the refresh in.